### PR TITLE
add unit tests for assets endpoints

### PR DESCRIPTION
**Pull request by B-Naoufal**

### CODE REVIEW

File: internal/handler/wallet/add_assets_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	// Get the handler for the AddAsset endpoint
+	handler, exist := api.HandlerFor("post", "/api/accounts/{nickname}/assets")
+	assert.True(t, exist)
+
+	// Send the POST request with an invalid address
+	resp, err := handleHTTPRequest(handler, "POST", fmt.Sprintf("/api/accounts/%s/assets?assetAddress=%s", nickname, invalidAddress), "")
+	assert.NoError(t, err)
+
+	// Assert that the response status code is 422 Unprocessable Entity
+	assert.Equal(t, http.StatusUnprocessableEntity, resp.Result().StatusCode)
```
Comment (Thykof) [inline internal/handler/wallet/add_assets_test.go:35]: check if response code 422 is present in the swagger, add it if no

File: internal/handler/wallet/add_assets_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+	"github.com/massalabs/station-massa-wallet/api/server/models"
+	utils "github.com/massalabs/station-massa-wallet/pkg/assets"
+	"github.com/massalabs/station-massa-wallet/pkg/wallet"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddAssetHandler_InvalidAddress(t *testing.T) {
+	api, _, _, _, err := MockAPI()
+	assert.NoError(t, err)
```
Comment (Thykof) [inline internal/handler/wallet/add_assets_test.go:17]: we usually do like this:

```
	api, _, _, _, err := MockAPI()
	assert.NoError(t, err)

	// test empty configuration first.
	t.Run("Get empty list", func(t *testing.T) {
		resp, err := processHTTPRequest(api, "GET", "/api/accounts", "")
		assert.NoError(t, err)
```

to not duplicate the creation of the API with `MockAPI`

File: internal/handler/wallet/get_all_assets_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	t.Helper()
+
+	handler, exist := api.HandlerFor("delete", "/api/accounts/{nickname}/assets")
+	assert.True(t, exist)
+
+	// Send the DELETE request to delete the asset
+	resp, err := handleHTTPRequest(handler, "DELETE", fmt.Sprintf("/api/accounts/%s/assets?assetAddress=%s", nickname, assetAddress), "")
+	assert.NoError(t, err)
+
+	assert.Equal(t, http.StatusCreated, resp.Result().StatusCode)
```
Comment (Thykof) [inline internal/handler/wallet/get_all_assets_test.go:162]: should be status code no content 204, not created 201

File: pkg/assets/asset.go [hunk 4]
```diff
@@ -262,3 +262,23 @@ func XMAInfo() models.AssetInfo {
 
 	return assetInfo
 }
+
+// RemoveJSONFile removes the assets.json file if it exists.
+func RemoveJSONFile() error {
```
Comment (Thykof) [inline pkg/assets/asset.go:267]: if it's only used in the tests, should we put this function in a test file? what is the use case of deleting the json assets file aside testing?

File: internal/handler/wallet/get_all_assets_test.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+	// Remove the json file created
+	err = utils.RemoveJSONFile()
+	assert.NoError(t, err)
+
+	err = cleanupTestData([]string{nickname})
+	assert.NoError(t, err)
+}
+
+func TestAddGetDeleteAsset(t *testing.T) {
```
Comment (Thykof) [inline internal/handler/wallet/get_all_assets_test.go:56]: this test case must be in `delete_asset_test.go`

File: internal/handler/wallet/get_all_assets_test.go [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+	// Parse the response body to get the assetsWithBalance
+	var assetsWithBalance []*models.AssetInfoWithBalance
+	err = json.Unmarshal(resp.Body.Bytes(), &assetsWithBalance)
+	assert.NoError(t, err)
+
+	return assetsWithBalance
+}
+
+func addAssetTest(t *testing.T, api *operations.MassaWalletAPI, nickname, assetAddress string) *models.AssetInfo {
```
Comment (Thykof) [inline internal/handler/wallet/get_all_assets_test.go:132]: this function should be in `add_asset_test.go`

File: internal/handler/wallet/get_all_assets_test.go [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+	// Parse the response body to get the added asset
+	var addedAsset models.AssetInfo
+	err = json.Unmarshal(resp.Body.Bytes(), &addedAsset)
+	assert.NoError(t, err)
+
+	return &addedAsset
+}
+
+func deleteAssetTest(t *testing.T, api *operations.MassaWalletAPI, nickname, assetAddress string) {
```
Comment (Thykof) [inline internal/handler/wallet/get_all_assets_test.go:152]: this function should be in `delete_asset_test.go `

**Review by Thykof — APPROVED** [review]

**Review by Thykof — CHANGES_REQUESTED** [review]
see my review above

File: internal/handler/wallet/assets.json
Comment (Thykof) [inline internal/handler/wallet/assets.json:1]: remove this, add this path in gitignore

File: internal/handler/wallet/add_assets_test.go
Comment (Thykof) [inline internal/handler/wallet/add_assets_test.go:1]: there is no passing test case here, this is strange

File: internal/handler/wallet/get_all_assets_test.go [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+	// Remove the json file created
+	err = RemoveJSONFile()
+	assert.NoError(t, err)
+
+	err = cleanupTestData([]string{nickname})
+	assert.NoError(t, err)
+}
+
+func getAssets(t *testing.T, api *operations.MassaWalletAPI, nickname string) []*models.AssetInfoWithBalance {
```
Comment (Thykof) [inline internal/handler/wallet/get_all_assets_test.go:55]: you can reuse this function in the test case above

**Review by Thykof — CHANGES_REQUESTED** [review]

File: internal/handler/wallet/add_assets_test.go
Comment (B-Naoufal) [reply internal/handler/wallet/add_assets_test.go:1]: it's covered in TestAddGetDeleteAsset

File: internal/handler/wallet/assets.json
Comment (Thykof) [reply internal/handler/wallet/assets.json:1]: do we need this path in gitignore?

Comment (B-Naoufal) [reply internal/handler/wallet/assets.json:1]: Done

File: internal/handler/wallet/get_all_assets_test.go [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+	assert.NoError(t, err)
+
+	nickname := "GoodNickname"
+	password := "zePassword"
+
+	_, errGenerate := wallet.Generate(nickname, password)
+	assert.Nil(t, errGenerate)
+
+	// Get the assetsWithBalance
+
```
Comment (Thykof) [inline internal/handler/wallet/get_all_assets_test.go:26]:
```suggestion
```

**Review by Thykof — APPROVED** [review]

**Review by peterjah — APPROVED** [review]

**Pull request merged**